src/storage.ts

```diff
@@ -290,10 +290,14 @@ export class Storage extends Service {
    * @param {StorageOptions} [options] Configuration options.
    */
   constructor(options: StorageOptions = {}) {
+    let url = `https://${options.apiEndpoint}/storage/v1`
```

Comment:
This is using `options.apiEndpoint` before it's set.  We should move up this line first:
```
 options.apiEndpoint = options.apiEndpoint || 'www.googleapis.com';

Reply:
Done